### PR TITLE
Patch #6712 - FieldCheckbox attribute default

### PR DESCRIPTION
**Pull request by Devportobello**
Checkbox field does not allow defining a value without being checked
#6712

For easiest test use this custom component: https://github.com/Devportobello/joomla-attach-PR-6794
## Test

First sry for my bad english, will try my best.

Dunno if we can test with something actually in Joomla, so if you want you can install my custom component then go to frontend "www.yourdomain.com/?option=com_test"
- _"with default" mean the attribute "value" of checkbox are set with something different that 1_
- _"with group" mean the use of fields in xml form definition_

**Test1**
Uncheck all checkbox then submit
-> Checkbox 1 with default and checkbox 2 with default are checked

**Test2**
Patch with the fix,
Uncheck all checkbox then submit
-> All checkbox are unchecked
## DEV

JForm::loadField
-> $field->setup($element, $value, $group) with value = $value or $default
Then
JFormFieldCheckbox::setup
-> empty($this->value) || $this->checked ? null : $this->checked = true;

### CODE REVIEW

File: libraries/joomla/form/fields/checkbox.php [hunk 1]
```diff
@@ -96,6 +96,15 @@ public function __set($name, $value)
 	 */
 	public function setup(SimpleXMLElement $element, $value, $group = null)
 	{
+		// Handle the default attribute
+		$default = (string) $element['default'];
+		if ($default)
```
Comment (zero-24) [inline libraries/joomla/form/fields/checkbox.php]: @Devportobello can you add a clean line befor the `if`

**Comment by zero-24** [comment]
@Devportobello sorry I'm not exactly sure what I need to test? Can you provide a step to step non technical testinstruktion? Thanks :smile:

**Comment by Devportobello** [comment]
First sry for my bad english, will try my best.

Dunno if we can test with something actually in Joomla, so if you want you can install my custom component then go to frontend "www.yourdomain.com/?option=com_test"
- *"with default" mean the attribute "value" of checkbox are set with something different that 1*
- *"with group" mean the use of fields in xml form definition*

**Test1**
Uncheck all checkbox then submit
-> Checkbox 1 with default and checkbox 2 with default are checked

**Test2**
Patch with the fix,
Uncheck all checkbox then submit
-> All checkbox are unchecked

**Comment by bertmert** [comment]
The goal is to have a checkbox that not just sends a value 0 or 1 if it's un/checked but an own value like *somevalue*.

Test instructions:
- Download com_test at https://github.com/Devportobello/joomla-attach-PR-6794
- Install it.
- Go to Frontend home. Add ?option=com_test.
- You'll see 2 checkboxes already checked just because the form XML file has set default values to them like
```
default="somevalue"
```
. Expected: No checkbox is checked when you open com_test first time.
- Uncheck both checkboxes. Check all other checkboxes. Click button.
- All checkboxes are checked after reload but only 2 should be checked.

- Apply patch: Test again.


**But see also comment** by @infograf768
https://github.com/joomla/joomla-cms/issues/6712#issuecomment-91789953
that fixes https://github.com/joomla/joomla-cms/issues/6712 by using field type checkboxes instead of field type checkbox.

**Comment by RonakParmar** [comment]
I have tested this item :white_check_mark: successfully on a3eb6b053cc1b6fe15a62cee8bc788bb96d62183<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6794">issues.joomla.org/joomla-cms/6794</a>.</sub>

**Comment by conconnl** [comment]
I have tested this item :white_check_mark: successfully on a3eb6b053cc1b6fe15a62cee8bc788bb96d62183<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6794">issues.joomla.org/joomla-cms/6794</a>.</sub>

**Comment by zero-24** [comment]
RTC based on tests. Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6794">issues.joomla.org/joomla-cms/6794</a>.</sub>

**Pull request merged**